internal/telemetry/telemetry.go

```diff
@@ -43,26 +45,59 @@ type Event struct {
 	Properties map[string]interface{} `json:"properties"`
 }
 
+type telemetryContextKey struct{}
+
+type telemetryContextValue struct {
+	startTime time.Time
+}
+
+func NewContext() context.Context {
+	return context.WithValue(context.Background(), contextKey, telemetryContextValue{
+		startTime: time.Now(),
+	})
+}
+
+func valueFromContext(ctx context.Context) (telemetryContextValue, bool) {
+	value, ok := ctx.Value(contextKey).(telemetryContextValue)
+	return value, ok
```

Comment:
```suggestion
	return ctx.Value(contextKey).(telemetryContextValue)
```

Reply:
see my comment above, I don't think this is needed

Reply:
![image](https://user-images.githubusercontent.com/2392468/166656638-0fad0b92-f527-4c98-beb5-bdb553c81ae6.png)